strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/repository/MavenGroupRepositoryComponentTest.java

```diff
@@ -0,0 +1,330 @@
+
+        String path = "com/artifacts/to/delete/releases/delete-group/1.2.1/delete-group-1.2.1.jar";
+        File artifactFile = new File(repository.getBasedir(), path);
+
+        assertTrue("Failed to locate artifact file " + artifactFile.getAbsolutePath(), artifactFile.exists());
+
+        layoutProvider.delete(STORAGE0, REPOSITORY_LEAF_L, path, false);
+        ((Maven2LayoutProvider) layoutProvider).closeIndex(STORAGE0, REPOSITORY_LEAF_L, path);
+
+        assertFalse("Failed to delete artifact file " + artifactFile.getAbsolutePath(), artifactFile.exists());
+
+
+        // author of changes
+        Metadata metadata = mavenMetadataManager.readMetadata(
+                layoutProvider.resolve(createRepositoryMock(STORAGE0, REPOSITORY_LEAF_L)).resolve(
+                        "com/artifacts/to/delete/releases/delete-group"));
+        assertThat(metadata.getVersioning().getVersions().size(), CoreMatchers.equalTo(1));
+        assertThat(metadata.getVersioning().getVersions().get(0), CoreMatchers.equalTo("1.2.2"));
+
+        // direct parent
+        metadata = mavenMetadataManager.readMetadata(
+                layoutProvider.resolve(createRepositoryMock(STORAGE0, REPOSITORY_GROUP_F)).resolve(
+                        "com/artifacts/to/delete/releases/delete-group"));
+        assertThat(metadata.getVersioning().getVersions().size(), CoreMatchers.equalTo(1));
+        assertThat(metadata.getVersioning().getVersions().get(0), CoreMatchers.equalTo("1.2.2"));
+
+        // next direct parent
+        metadata = mavenMetadataManager.readMetadata(
+                layoutProvider.resolve(createRepositoryMock(STORAGE0, REPOSITORY_GROUP_B)).resolve(
+                        "com/artifacts/to/delete/releases/delete-group"));
+        assertThat(metadata.getVersioning().getVersions().size(), CoreMatchers.equalTo(1));
+        assertThat(metadata.getVersioning().getVersions().get(0), CoreMatchers.equalTo("1.2.2"));
+
+        // grand parent
+        metadata = mavenMetadataManager.readMetadata(
+                layoutProvider.resolve(createRepositoryMock(STORAGE0, REPOSITORY_GROUP_H)).resolve(
+                        "com/artifacts/to/delete/releases/delete-group"));
+        assertThat(metadata.getVersioning().getVersions().size(), CoreMatchers.equalTo(1));
+        assertThat(metadata.getVersioning().getVersions().get(0), CoreMatchers.equalTo("1.2.2"));
+
+        // grand parent with other kids
+        metadata = mavenMetadataManager.readMetadata(
+                layoutProvider.resolve(createRepositoryMock(STORAGE0, REPOSITORY_GROUP_A)).resolve(
+                        "com/artifacts/to/delete/releases/delete-group"));
+        assertThat(metadata.getVersioning().getVersions().size(), CoreMatchers.equalTo(2));
+        assertThat(metadata.getVersioning().getVersions().get(0), CoreMatchers.equalTo("1.2.1"));
+        assertThat(metadata.getVersioning().getVersions().get(1), CoreMatchers.equalTo("1.2.2"));
+    }
+
+    /*
```

Comment:
Is this work-in-progress (and is therefore commented out), or is it something that will be removed?

Reply:
garbage :)